powerlevel9k.zsh-theme

```diff
@@ -1628,11 +1628,21 @@ prompt_virtualenv() {
 }
 
 ################################################################
-# pyenv: current active python version (with restrictions)
+# Segment to display pyenv information
 # https://github.com/pyenv/pyenv#choosing-the-python-version
+set_default POWERLEVEL9K_PYENV_PROMPT_ALWAYS_SHOW false
 prompt_pyenv() {
-  if [[ -n "$PYENV_VERSION" ]]; then
-    "$1_prompt_segment" "$0" "$2" "blue" "$DEFAULT_COLOR" "$PYENV_VERSION" 'PYTHON_ICON'
+  if ! [ -x "$(command -v pyenv)" ]; then
+    if [[ -n "$PYENV_VERSION" ]]; then
+      "$1_prompt_segment" "$0" "$2" "blue" "$DEFAULT_COLOR" "$PYENV_VERSION" 'PYTHON_ICON'
+    else
```

Comment:
That could be an `elif ! [ -x "$(command -v pyenv)" ]; then`. That way we could save the outer if.. I see that you've done it because the `$PYENV_VERSION` should only be there, if `pyenv` is there too, but IMHO if `$PYENV_VERSION` is set, we should output that information as quickly as possible.

Reply:
What if I incorrectly used a `.zshrc` file including something like `export PYENV_VERSION=3.5.5` without `pyenv` installed?

Reply:
What if someone installed `pyenv`, but never calls it and set `PYENV_VERSION=1.2.3`? What if someone wrote a script and called it `pyenv`?

I think there are always scenarios we cannot handle. That is why I would rely on the variable in the first place..

Reply:
Fine, maybe I thought too much. :smile: